editor/history.js

```diff
@@ -130,16 +153,9 @@ export class MoveElementCommand extends Command {
    * @returns {void}
   */
   apply (handler) {
-    // TODO(codedread): Refactor this common event code into a base HistoryCommand class.
-    if (handler) {
-      handler.handleHistoryEvent(HistoryEventTypes.BEFORE_APPLY, this);
-    }
-
-    this.elem = this.newParent.insertBefore(this.elem, this.newNextSibling);
-
-    if (handler) {
-      handler.handleHistoryEvent(HistoryEventTypes.AFTER_APPLY, this);
-    }
+    super.apply(handler, () => {
+      this.elem = this.newParent.insertBefore(this.elem, this.newNextSibling);
```

Comment:
i.e., I think it may be:

```js
this.newNextSibling.before(this.elem);
this.elem = this.newNextSibling;
```

Reply:
insertBefore (https://developer.mozilla.org/en-US/docs/Web/API/Node/insertBefore) returns the newNode (i.e the first argument which is in our case this.elem). I believe we don't need to add the second line and it would just work. Am I correct?

Reply:
I'm merging as it is now but we can indeed improve this later.

Reply:
Yes, `insertBefore` works as you have it.

But I think the new APIs (i.e., in this case `before`) dropped the old behavior of returning a Node as it was a kind of magic that is less than immediately clear about what is being returned.

(FWIW, This preference is normally enforced by a Unicorn linting rule but in this case, it doesn't happen to find them at the moment when `insertBefore` has more complex contents (i.e., if you care to know the details, it would report `insertBefore(a, b) with simple variable identifier arguments, but not if the args had (dotted) member expressions like `insertBefore(this.elem, this.newNextSibling)`).)

Btw, I've updated Travis as it wasn't properly executing the linting. It is now failing due to a few console statements, so if you could either remove or eslint-disable those statements, that should allow the builds to pass.

Reply:
The one line I was mentioning is `this.newNextSibling.before(this.elem);`

Reply:
If you mean that my change to add `this.newNextSibling.before(this.elem);` should be sufficient, no, because `before` only returns `undefined`. I therefore did need to add the extra line: `this.elem = this.newNextSibling;`, even though `insertBefore`, which did return a usable Node, did not need an extra line.

While two lines is generally worse, I have come to believe there is wisdom in the API change in *not* returning a value, and forcing one to add another line.

Reply:
Let me clarify: 
if you don't assign the result of this.newNextSibling.before(this.elem) to this.elem i.e. 
```suggestion
      this.newParent.insertBefore(this.elem, this.newNextSibling);
```
then this.elem does not need to be reassigned.

Reply:
Ah, yes, good point. Cool--even better!

Reply:
Oh, and sorry, re: the Travis update to enforce linting, I realized it hadn't pushed, but have now done so.